conda/resolve.py

```diff
@@ -843,7 +843,7 @@ def push_MatchSpec(self, C, spec):
             libs = [fkey for fkey in tgroup if spec.match(fkey)]
         if len(libs) == len(tgroup):
             if spec.optional:
-                m = True
+                m = TRUE
```

Comment:
..... and this kids is how you waste more than an hour of time...
(Meaning: It was "real fun" when I realized the changes to `logic.py` were erroneous just because I forgot to check the input here.)